vacasa/connect/connect.py

```diff
@@ -488,6 +488,48 @@ def get_unit_amenity_properties(self, params: dict = None):
 
         return self._iterate_pages(url, headers, params)
 
+    def add_unit_amenity_property(self,
+                                  amenity_id: int,
+                                  unit_id: int,
+                                  idamenities_property: int,
+                                  property_value: str) -> dict:
+        """Add a new unit amenity property in Connect.
+
+        Args:
+            amenity_id: An amenity ID.
+            unit_id: A unit ID.
+            idamenities_property: An amenity property ID.
+            property_value: The property value for the new unit amenity property.
+
+        Returns:
+            json response for success
+        """
+        url = f"{self.endpoint}/v1/unit-amenity-properties"
+        headers = self._headers()
+        payload = {
+            'amenity_id': amenity_id,
+            'unit_id': unit_id,
+            'idamenities_property': idamenities_property,
+            'property_value': property_value
+        }
+
+        return self._post(url, json={'data': {'attributes': payload,
+                                              'type': 'unit-amenity-property'}}, headers=headers).json()
```

Comment:
This is just personal preference, but I think this would be more readable:
```python
payload = {
    'data': {
        'type': 'unit-amenity-property',
        'attributes': {
            'amenity_id': amenity_id,
            'unit_id': unit_id,
            'idamenities_property': idamenities_property,
            'property_value': property_value
        }
    }
}

return self._post(url, json=payload, headers=headers).json()
```

Reply:
It also might not be a bad idea to check for success before trying to parse the JSON response.
```python
r = self._post(url, json=payload, headers=headers)
r.raise_for_status()
return r.json()
```

Reply:
Okay, I'll add these two suggestions.

Reply:
I'd merged the code so these changes are in https://github.com/Vacasa/python-vacasa-connect-sdk/pull/86.